oauth2/oauth2-resource-server/src/test/java/org/springframework/security/oauth2/resourceserver/web/BearerTokenAuthenticationEntryPointTests.java

```diff
@@ -0,0 +1,192 @@
+/*
+ * Copyright 2002-2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.security.oauth2.resourceserver.web;
+
+import java.io.IOException;
+
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+import org.springframework.http.HttpStatus;
+import org.springframework.mock.web.MockHttpServletRequest;
+import org.springframework.mock.web.MockHttpServletResponse;
+import org.springframework.security.authentication.BadCredentialsException;
+import org.springframework.security.oauth2.resourceserver.BearerTokenAuthenticationException;
+import org.springframework.security.oauth2.resourceserver.BearerTokenError;
+import org.springframework.security.oauth2.resourceserver.BearerTokenErrorCodes;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+/**
+ * Tests for {@link BearerTokenAuthenticationEntryPoint}.
+ *
+ * @author Vedran Pavic
+ */
+public class BearerTokenAuthenticationEntryPointTests {
+
+	@Rule
+	public ExpectedException thrown = ExpectedException.none();
+
+	private BearerTokenAuthenticationEntryPoint authenticationEntryPoint;
+
+	@Before
+	public void setUp() {
+		this.authenticationEntryPoint = new BearerTokenAuthenticationEntryPoint();
+	}
+
+	@Test
+	public void commence_AuthenticationException_ShouldSend401AndHeader() throws IOException {
```

Comment:
I've named them like that in #3 however this seems approach seems much more readable (also see [this article](https://www.petrikainulainen.net/programming/testing/writing-clean-tests-naming-matters/)). I have no problem with reverting naming, if someone really doesn't like this approach.

Reply:
I would guess that it would come down to any agreed upon standard. I just double-checked the contribution guidelines, and I don't see anything there either way (only that it shouldn't start with "test"). So, if we have a standard, it's only been communicated internally at this point.